Bots/DotNet/Consumers/CodeFirst/SimpleHostBot-2.1/Bots/HostBot.cs

```diff
@@ -141,30 +120,102 @@ protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersA
+            {
+                eocActivityMessage += $"\n\nText: {activity.Text}";
+            }
+
+            if (activity.Value != null)
+            {
+                eocActivityMessage += $"\n\nValue: {JsonConvert.SerializeObject(activity.Value)}";
+            }
+
+            await turnContext.SendActivityAsync(MessageFactory.Text(eocActivityMessage), cancellationToken);
+
+            // We are back at the host.
+            await turnContext.SendActivityAsync(MessageFactory.Text("Back in the host bot."), cancellationToken);
+        }
+
         /// <summary>
         /// Sends an activity to the skill bot.
         /// </summary>
         /// <param name="turnContext">Context for the current turn of conversation.</param>
         /// <param name="targetSkill">The skill that will receive the activity.</param>
         /// <param name="cancellationToken">CancellationToken propagates notifications that operations should be cancelled.</param>
         /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
-        private async Task SendToSkillAsync(ITurnContext<IMessageActivity> turnContext, BotFrameworkSkill targetSkill, CancellationToken cancellationToken)
+        private async Task SendToSkillAsync(ITurnContext<IMessageActivity> turnContext, string deliveryMode, BotFrameworkSkill targetSkill, CancellationToken cancellationToken)
         {
             // NOTE: Always SaveChanges() before calling a skill so that any activity generated by the skill
             // will have access to current accurate state.
             await _conversationState.SaveChangesAsync(turnContext, force: true, cancellationToken: cancellationToken);
 
-            // route the activity to the skill
-            var response = await _skillClient.PostActivityAsync(_botId, targetSkill, _skillsConfig.SkillHostEndpoint, (Activity)turnContext.Activity, cancellationToken);
+            // Clone activity and update its delivery mode.
+            var activity = JsonConvert.DeserializeObject<Activity>(JsonConvert.SerializeObject(turnContext.Activity));
+            activity.DeliveryMode = deliveryMode;
 
-            // Check response status
-            if (!(response.Status >= 200 && response.Status <= 299))
+            switch (deliveryMode)
             {
-                throw new HttpRequestException($"Error invoking the skill id: \"{targetSkill.Id}\" at \"{targetSkill.SkillEndpoint}\" (status is {response.Status}). \r\n {response.Body}");
+                case DeliveryModes.ExpectReplies:
+                    // Route the activity to the skill.
+                    var expectRepliesResponse = await _skillClient.PostActivityAsync<ExpectedReplies>(_botId, targetSkill, _skillsConfig.SkillHostEndpoint, activity, cancellationToken);
+
+                    // Check response status.
+                    if (!(expectRepliesResponse.Status >= 200 && expectRepliesResponse.Status <= 299))
```

Comment:
Consider using IsSuccessStatusCode here (I don't think this is available on JS or Python but it makes things simpler in C#)
```suggestion
                    if (!expectRepliesResponse.IsSuccessStatusCode())
```